pkg/antctl/raw/traceflow/command.go

```diff
@@ -233,90 +216,71 @@ func newTraceflow(client kubernetes.Interface) (*v1alpha1.Traceflow, error) {
 	return tf, nil
 }
 
-func dstIsPod(client kubernetes.Interface, ns string, name string) (bool, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
-	defer cancel()
-	_, err := client.CoreV1().Pods(ns).Get(ctx, name, metav1.GetOptions{})
+func parseFlow() (*v1alpha1.Packet, error) {
+	cleanFlow := strings.ReplaceAll(option.flow, " ", "")
+	fields, err := getPortFields(cleanFlow)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return false, nil
-		}
-		return false, fmt.Errorf("failed to get Pod from Kubernetes API: %w", err)
+		return nil, fmt.Errorf("error when parsing the flow: %w", err)
 	}
-	return true, nil
-}
 
-func parseFlow() (*v1alpha1.Packet, error) {
-	cleanFlow := strings.ReplaceAll(option.flow, " ", "")
 	pkt := new(v1alpha1.Packet)
-
-	for _, v := range strings.Split(cleanFlow, ",") {
-		n, ok := protocols[v]
-		if ok {
-			(*pkt).IPHeader.Protocol = n
+	for k, v := range protocols {
+		if _, ok := fields[k]; ok {
+			pkt.IPHeader.Protocol = v
 			break
 		}
 	}
 
-	r, ok, err := getFieldPortValue(cleanFlow, "tcp_src")
-	if err != nil {
-		return nil, fmt.Errorf("error when get tcp_src value: %w", err)
-	}
-	if ok {
-		if (*pkt).TransportHeader.TCP == nil {
-			(*pkt).TransportHeader.TCP = new(v1alpha1.TCPHeader)
+	if r, ok := fields["tcp_src"]; ok {
```

Comment:
I think you can initialize `pkt.TransportHeader.TCP` or `pkt.TransportHeader.UDP` after L229, then skip all following nil check.

Reply:
To save memory, skip nil check in "tcp_src" and "udp_src".